### PR TITLE
perf(sql): parallelise sample and population standard deviation

### PR DESCRIPTION
**Pull request by nwoolmer**
`stddev`, `stddev_samp`, `stddev_pop`

First 1 million rows of `trade` demo data.

Before and after:

```sql

SELECT stddev_samp(price) from trades where symbol = 'ETH-USD'
-- gives 75.471371580724
-- takes 8-9 ms

SELECT stddev_samp(price) from trades where symbol = 'ETH-USD'
-- gives 75.471371580... (3 digit instability)
-- takes 3-4 ms

SELECT stddev_pop(price) from trades where symbol = 'ETH-USD'
-- gives 75.471297621052
-- takes 8-9 ms

SELECT stddev_pop(price) from trades where symbol = 'ETH-USD'
-- gives 75.471297621... (3 digit instability)
-- takes 3-4 ms
```

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:disappointed: **fail** : 20 / 48 (41.67%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/functions/groupby/StdDevSampleGroupByFunctionFactory.java|10|24|41.67%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/groupby/StdDevPopGroupByFunctionFactory.java|10|24|41.67%|

**Review by ideoma — COMMENTED** [review]
Please add test cases to ParallelGroupByFuzzTest, otherwise the merge is not covered.

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 48 / 48 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/functions/groupby/StdDevSampleGroupByFunctionFactory.java|24|24|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/groupby/StdDevPopGroupByFunctionFactory.java|24|24|100.00%|

File: core/src/main/java/io/questdb/griffin/engine/functions/groupby/StdDevPopGroupByFunctionFactory.java [hunk 1]
```diff
@@ -46,4 +49,60 @@ public boolean isGroupBy() {
...
+
+            long mergedCount = srcCount + destCount;
+            double delta = destMean - srcMean;
+
+            // This is only valid when countA is much larger than countB.
+            // If both are large and similar sizes, delta is not scaled down.
+            // double mergedMean = srcMean + delta * ((double) destCount / mergedCount);
+
+            // So we use this instead:
+            double mergedMean = (srcCount * srcMean + destCount * destMean) / mergedCount;
```
Comment (ideoma) [inline core/src/main/java/io/questdb/griffin/engine/functions/groupby/StdDevPopGroupByFunctionFactory.java:94]: Why do we need to calculate the mean here? It's not used in `getDouble()`, only sum and count are used.
If you don't need the mean here, there is also no point calculating it in the base class for this implementation.

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 48 / 48 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/functions/groupby/StdDevSampleGroupByFunctionFactory.java|24|24|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/groupby/StdDevPopGroupByFunctionFactory.java|24|24|100.00%|

**Review by bluestreak01 — APPROVED** [review]

**Pull request merged**